bazel/api-gen/manifest/index.ts

```diff
@@ -0,0 +1,35 @@
+import {readFileSync, writeFileSync} from 'fs';
+
+/** The JSON data file format for extracted API reference info. */
+interface EntryCollection {
+  moduleName: string;
+  entries: {name: string; entryType: string}[];
+}
+
+export interface ManifestEntry {
+  name: string;
+  type: string;
+}
+
+/** Manifest that maps each module name to a list of API symbols. */
+type Manifest = Record<string, ManifestEntry[]>;
+
+function main() {
+  const [srcs, outputFilenameExecRootRelativePath] = process.argv.slice(2);
+
+  const sourceContents = srcs
+    .split(',')
+    .map((srcPath) => readFileSync(srcPath, {encoding: 'utf8'}));
```

Comment:
You could do this "in parallel" with promises.


```suggestion
  const sourceContents = await Promise.all(srcs
    .split(',')
    .map((srcPath) => readFile(srcPath, {encoding: 'utf8'}));
```

Reply:
My understanding is that the sync APIs are actually faster because node is single threaded, so the overhead of managing the async operations just adds overhead (though it's been years since I've internalized, not sure if there's new info I don't have)

Reply:
Okay, works for me